### ISSUE TITLE
Revise Frog Enclosure Mounts for Raspberry Pi CM4

### ISSUE THREAD

**Issue by keenanjohnson** [Ribbit-Network/ribbit-network-frog-sensor#132]
In #130 we identified that we need to go back to a Raspberry Pi CM4 version of the Frog Sensor.

This task is to revise the mechanical enclosure to fit the Raspberry Pi CM4 and Carrier Board as well as Antennae. 

An older version of the Onshape Assembly has the correct parts and geometry I believe, so we will just need to bring those further into the newer version of the assembly that has the better sensors at the top of the Frog.

V5 in OnShape: 
![image](https://user-images.githubusercontent.com/2559382/180928623-1e955387-5b4b-458e-ad3d-f72605ce6934.png)

**Comment by eaudiffred** [comment]
@keenanjohnson, A couple of questions for you if you don't mind.
1. I'm seeing different port locations on the Raspberry Pi board that I look up on the internet and the one that you have and used in V1.  Is the board you are using some kind of modified version?  The power port is in a much better location on the version shown above, but I can't seem to find that model.
2. In OnShape I took a stab at creating a branch, but I may have messed it up, sorry!  Should I continue working on the light Blue or Purple branch?

**Comment by eaudiffred** [comment]
Oh, just saw your reply in Discord!
_"We actually used the Raspberry Pi CM4 with a different carrier board which is slightly different than the Raspberry Pi 4. We'll also need to find a spot for the external antennae."_

- What is the brand or model of the carrier board?  I'd like to find a dimension drawing.

- Which antenna needs to be used with the Pi?  The black on shown above, or the Adafruit one (https://www.adafruit.com/product/2308)?

**Comment by keenanjohnson** [comment]
Yes great questions!

Here are the parts and links to datasheets!

* [Waveshare Mini Base Board (A) Designed for Raspberry Pi Compute Module 4](https://www.waveshare.com/product/raspberry-pi/boards-kits/compute-module-4-cat/cm4-io-base-a.htm)
* [CM4102000 Raspberry Pi Compute Module 4 Lite](https://www.newark.com/raspberry-pi/cm4102000/rpi-compute-module-4-lite-2gb/dp/86AH2093)
* [Raspberry Pi CM4 Antennae](https://datasheets.raspberrypi.com/cm4/cm4-antenna-kit-product-brief.pdf)

**Comment by keenanjohnson** [comment]
One of the tradeoffs that I think needs to be made @eaudiffred is whether the antennae is going to live inside or outside the enclosure. The antennae is technically water resistant, however, it doesn't have an official IP rating or anything.

I had played around with it either way, but I'm not sure if you have opinions or thoughts @eaudiffred ?

External Ant. Example:

<img width="603" alt="image" src="https://user-images.githubusercontent.com/2559382/181074188-31ae7324-c60b-4db2-a8bf-acf8846719b7.png">

Internal Ant. Example:

![image](https://user-images.githubusercontent.com/2559382/181074584-93af71ca-0cbf-4427-8daf-3aca626f42e1.png)

**Comment by eaudiffred** [comment]
@keenanjohnson, Thanks for the links! 

In my opinion, inside the enclosure is preferred for these reasons:

1. Less holes in the enclosure keeps external things out (rain, bugs, dust)
2. It presents a cleaner look
3. Over all footprint is reduced

However, if it compromises the range of the antenna, forces a much larger enclosure, or makes assembly overly difficult, I think having it stick out would be fine.  The cable and connections are protected either way.  

Do you know if the range is reduced if its housed inside?

**Comment by keenanjohnson** [comment]
No the PETG of the enclosure shouldn't compromise the signal integrity as I understand it, so this is just a mechanical packaging issue :). So sounds like inside it is!

**Comment by eaudiffred** [comment]
Cool.  Which branch should I work off of in OnShape?  I see you made one branched from an earlier bbgg edition.  Should I continue with that one?

**Comment by keenanjohnson** [comment]
Oh I would work in the one you made! That has all the latest enclosure
stuff. Those other branches are are old from me, I just couldn't figure out
a way to delete them haha.

**Comment by keenanjohnson** [comment]
No rush @eaudiffred , but just making sure you have everything you need here?

**Comment by eaudiffred** [comment]
@keenanjohnson, yep.  Working on the antenna now.  I think that's the last change I need to make.

**Comment by eaudiffred** [comment]
I think it's ready.  Had to extend the length of the enclosure and lid by 15mm.  I think that's an ok amount to keep it all enclosed.  And if I remember correctly we did shorten it when we switched to the BBGG board, so now it's back to the original.  

![image](https://user-images.githubusercontent.com/88562047/181416702-3534933f-05cb-496d-9a92-776dd675491a.png)

**Comment by eaudiffred** [comment]
@keenanjohnson let me know if you see any glaring errors or parts that aren't fitting.  I have a head on my printer right now, but I'll try and get a base and lid going tomorrow.

**Comment by keenanjohnson** [comment]
Sounds good @eaudiffred! We don't actually have to account for the USB cable at the top there any longer since we switched GPS units, but it looks like the ant. length is driving the height anyway. 

If we ditched the USB in the model, bent the ant. and moved it up and upside down, perhaps we could save some packaging space? Not sure if it is really worth it though. 

I'm thinking something like this:

![IMG_3689](https://user-images.githubusercontent.com/2559382/181438581-2a1e78b4-5300-454a-8fe4-95492e27a876.JPG)

This is definitely workable though and I'll try a print tonight and can fit check parts as well!

**Comment by eaudiffred** [comment]
@keenanjohnson, sweet.  I actually had it modeled like that, but then changed it because of the usb connection, lol.  Bending it actually works pretty well.  I'll rework it and put in the bend.  Does anything need to be plugged into those top three ports?  That should save that 15mm too.

**Comment by eaudiffred** [comment]
@keenanjohnson this might be a dumb question, but does it make a difference if the antenna is mounted upside down or right side up?

**Comment by eaudiffred** [comment]
Here is option #1.
![image](https://user-images.githubusercontent.com/88562047/181520307-3ade069b-3022-4f94-9e33-dabe8a54857e.png)

**Comment by eaudiffred** [comment]
#2 would be flipping the antenna upside down.  Either way the antenna cable comes close to the entry/exit holes for the other cables, but I don't think it will cause a problem.

**Comment by keenanjohnson** [comment]
Option 2 seems slightly worse to me from a packaging perspective because the part of the wifi ant. cable nearest the big plastic part of the ant. is the most stiff, making it hard to not poke the wifi cable with the power cable when you plug it into power.


![IMG_3696](https://user-images.githubusercontent.com/2559382/181596445-c23cbde9-25ef-4e53-a116-eb926e54e75b.JPG)

For option 1, we can sort of route the cable around under the pi something like this:

![IMG_3689](https://user-images.githubusercontent.com/2559382/181601342-fae2af1f-abb0-4026-bcaa-3fcc0be5b04a.jpg)

**Comment by eaudiffred** [comment]
Updated!
![image](https://user-images.githubusercontent.com/88562047/181606877-3350f88d-40bb-4dc9-91d3-98df1b9b4017.png)

**Comment by keenanjohnson** [comment]
Thanks @eaudiffred ! I'll do a test print soon.

**Comment by keenanjohnson** [comment]
Hey @eaudiffred ! I had a chance to do the test print and put all the parts in for a fit check.

Overall, this seems pretty good and I like the way the cables will be routed:

![IMG_3791](https://user-images.githubusercontent.com/2559382/182782632-83736238-1848-42e5-81c4-b76628a4bf24.JPG)
![IMG_3792](https://user-images.githubusercontent.com/2559382/182782733-a218eeb5-9688-4e79-b5f2-653babc4b51c.JPG)

Unfortunately, there is a misalignment on the height of the raspberry pi and the hole for the power cable as you can see below. Using a rule, my best estimate is that the hole probably needs to come down about 4 mm or so and things should be aligned.
![IMG_3793](https://user-images.githubusercontent.com/2559382/182782850-859e01cf-a8dc-49f5-b716-f03283c856f3.JPG)

**Comment by keenanjohnson** [comment]
Once that's resolved, things should look nice.

![IMG_3794](https://user-images.githubusercontent.com/2559382/182783064-1d86d96e-880c-4e28-b62e-0743e016ae22.JPG)

**Comment by eaudiffred** [comment]
@keenanjohnson, Excellent, glad the antenna fits.  I realize in the CAD assembly model the board is sitting on brass posts that are screwed into the plastic posts that we designed into the base.  That's why the hole for the power cord is off.  I'll bump down the hole and remove the brass from the assembly.  I should be able to knock it out tomorrow.

**Comment by eaudiffred** [comment]
Those couple of changes have been made.  @keenanjohnson how did the antenna mounting go?  Was the hole correctly sized and could you tighten the nut without too much difficulty?

**Comment by keenanjohnson** [comment]
Oh, I didn't realize the model had standoffs. I can also do those if you think they are important.

Oh yeah, the ant. mount seemed great! It's not exactly super easy to screw it in, but I think that is the ant. itself not the mount. I was thinking more about future improvements and perhaps in the future, maybe we could go to a clip-based system for the ant. instead? That's more of a future idea, perhaps the raspberry pi could be held in clips as well in the future, but perhaps that's not as robust as screws.

**Comment by eaudiffred** [comment]
Removing the standoff is one less thing for people to have to install, so I think it makes sense to get rid of them.  You can install them in the one you have printed though to salvage that enclosure.  

I like the clip idea, getting rid of hardware is a good thing.  I think its possible with printing to design something in.  

When would you like to call V3 complete?  Should we continue to modify and experiment for now, or are we ready to crank out 30 of them to go along with the Pis that you have?

**Comment by keenanjohnson** [comment]
Let's move the thought experiment of the clips to another issue and we can just pursue that in parallel as you have time to do the modeling. If you have time and it's ready to be part of V3 then that's great, but if not, no big deal.

**Comment by eaudiffred** [comment]
Sounds good.  Are there other items that need to be completed for the launch of V3 or was the enclosure the last part?  Just trying to get a feel for what kind of timeline you are shooting for to finalize V3.

**Comment by keenanjohnson** [comment]
Basically just this mechanical update and then updating the build instructions for v3 are left, so probably pretty quick. I hope to resolve v3 by mid-week maybe after I have time to do a test print / take new pictures for the instructions etc.

https://github.com/Ribbit-Network/ribbit-network-frog-sensor/milestone/3

**Comment by keenanjohnson** [comment]
This is complete! I think we just need to update the mechanical files in the v3_docs branch on git.

**Comment by keenanjohnson** [comment]
These files will be uploaded in #143

**Comment by eaudiffred** [comment]
Completed and implemented as part of the V3 changes.
